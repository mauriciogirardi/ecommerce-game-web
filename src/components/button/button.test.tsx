import { render, screen } from '@testing-library/react'
import { Badge } from 'lucide-react'
import { describe, expect, it } from 'vitest'

import { Button } from '.'

describe('<Button />', () => {
  it('should be able to render the button', () => {
    const { container } = render(<Button>click here</Button>)

    const button = screen.getByRole('button', { name: /click here/i })
    expect(button).toBeInTheDocument()
    expect(button).toHaveClass('bg-linear h-9 px-3 text-sm')

    expect(container).toMatchSnapshot()
  })

  it('should be able to render the button with size small', () => {
    render(<Button size="small">click here</Button>)

    const button = screen.getByRole('button', { name: /click here/i })
    expect(button).toHaveClass('h-8 px-2.5')
  })

  it('should be able to render the button with size large', () => {
    render(<Button size="large">click here</Button>)

    const button = screen.getByRole('button', { name: /click here/i })
    expect(button).toHaveClass('h-12 px-6')
  })

  it('should be able to render the button with icon', () => {
    render(<Button size="icon" icon={<Badge data-testid="icon" />} />)

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should be able to render the button disabled', () => {
    render(<Button>click here</Button>)
    const button = screen.getByRole('button', { name: /click here/i })
    expect(button).toHaveClass(
      'disabled:cursor-not-allowed disabled:opacity-50'
    )
  })
})
