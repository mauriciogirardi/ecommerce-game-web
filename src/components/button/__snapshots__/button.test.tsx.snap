// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should be able to render the button 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 bg-linear text-white hover:opacity-90 h-9 px-3 text-sm"
  >
    click here
  </button>
</div>
`;
